petastorm/fs_utils.py

```diff
@@ -134,8 +131,12 @@ def __init__(self, dataset_url, hadoop_configuration=None, connector=HdfsConnect
 
             storage_options = storage_options or {}
             protocol = self._parsed_dataset_url.scheme
-            self._filesystem = fsspec.filesystem(protocol, **storage_options)
-            self._filesystem_factory = lambda: fsspec.filesystem(protocol, **storage_options)
+            cls = fsspec.get_filesystem_class(protocol)
+            options = cls._get_kwargs_from_urls(self._dataset_url)
+            update_storage_options(options, storage_options)
+            fs = cls(**options)
+            self._filesystem = cls(**options)
```

Comment:
Why do we construct `cls` twice? (lines 138 and 137)

Reply:
typo. Will update the code.

Reply:
typo. Updated changes.